### PR TITLE
Implemented: Add shop selector to menu footer(#2a7ymj3)

### PR DESCRIPTION
**Pull request by Mayank909**
### Related Issues

Closes #

### Screenshots of Visual Changes before/after (If There Are Any)

**IMPORTANT NOTICE** - Remember to add changelog entry

### Contribution and Currently Important Rules Acceptance

- [ ] I read and followed [contribution rules](https://github.com/hotwax/job-manager#contribution-guideline)

https://user-images.githubusercontent.com/44576058/167864489-85820e45-a100-495d-9a0e-7ea231553b57.mp4

### CODE REVIEW

File: src/views/Pipeline.vue [hunk 1]
```diff
@@ -384,18 +384,27 @@ export default defineComponent({
         this.store.dispatch('job/fetchPendingJobs', {eComStoreId: this.getCurrentEComStore.productStoreId, viewSize:process.env.VUE_APP_VIEW_SIZE, viewIndex:0}).then(() => {
           if(event) event.target.complete();
           this.isRetrying = false;
+          emitter.off("selectedShop",this.refreshJobs);
+        }).catch(()=>{
```
Comment (ymaheshwari1) [inline src/views/Pipeline.vue:388]: Why we used catch here?

File: src/views/Pipeline.vue [hunk 2]
```diff
@@ -384,18 +384,27 @@ export default defineComponent({
...
+          this.isRetrying = false;
         });
       } else if(this.segmentSelected === 'running') {
         this.store.dispatch('job/fetchRunningJobs', {eComStoreId: this.getCurrentEComStore.productStoreId, viewSize:process.env.VUE_APP_VIEW_SIZE, viewIndex:0}).then(() => {
           if(event) event.target.complete();
           this.isRetrying = false;
+          emitter.off("selectedShop",this.refreshJobs);
```
Comment (ymaheshwari1) [inline src/views/Pipeline.vue:395]: We can define this emitter.off outside the if else block, so that we don't need to redeclare in each if block.

**Review by ymaheshwari1 — CHANGES_REQUESTED** [review]
We also need to handle this product store change functionality for all the other pages

File: src/views/Pipeline.vue (hunk 2 above)
Comment (Mayank909) [reply src/views/Pipeline.vue:395]: Done Sir

File: src/views/Pipeline.vue (hunk 1 above)
Comment (Mayank909) [reply src/views/Pipeline.vue:387]: Sir, When the promise is resolved It goes into the 'then' block. If 'then' returns a rejected promise, it goes to catch block. That's why I use the catch here.

File: src/views/Product.vue [hunk 3]
```diff
@@ -91,12 +91,10 @@ export default defineComponent({
     }
   },
   mounted () {
-    this.store.dispatch("job/fetchJobs", {
-      "inputFields":{
-        "systemJobEnumId": Object.values(this.jobEnums),
-        "systemJobEnumId_op": "in"
-      }
-    });
+    emitter.on("selectedShop", this.fetchJobs);
```
Comment (ymaheshwari1) [inline src/views/Product.vue:94]: As you have removed the fetchJobs action from mounted hook, then when the component is mounted we won't get the job data.

**Review by ymaheshwari1 — CHANGES_REQUESTED** [review]

File: src/views/Product.vue (hunk 3 above)
Comment (Mayank909) [reply src/views/Product.vue:94]: OK Sir I Will correct it.

File: src/components/Menu.vue [hunk 4]
```diff
@@ -86,6 +93,16 @@ export default defineComponent({
       userProfile: 'user/getUserProfile'
     })
   },
+  methods: {
+    setEComStore(store: any) {
```
Comment (ymaheshwari1) [inline src/components/Menu.vue:97]:
```suggestion
    setEComStore(event: CustomEvent) {
```

**Review by ymaheshwari1 — CHANGES_REQUESTED** [review]

File: src/components/Menu.vue [hunk 5]
```diff
@@ -86,6 +93,16 @@ export default defineComponent({
...
+      if(this.userProfile) {
+        this.store.dispatch('user/setEcomStore', {
+          'eComStore': this.userProfile.stores.find((str: any) => str.productStoreId == store['detail'].value)
```
Comment (ymaheshwari1) [inline src/components/Menu.vue:100]:
```suggestion
          'eComStore': this.userProfile.stores.find((store: any) => store.productStoreId == store['detail'].value)
```

File: src/components/Menu.vue [hunk 6]
```diff
@@ -86,6 +93,16 @@ export default defineComponent({
...
+        })
+      }
+      emitter.emit("selectedShop", store)
```
Comment (ymaheshwari1) [inline src/components/Menu.vue:103]: I think there is no need to pass store.

**Review by ymaheshwari1 — CHANGES_REQUESTED** [review]

File: src/components/Menu.vue (hunk 4 above)
Comment (Mayank909) [reply src/components/Menu.vue:97]: OK sir

File: src/components/Menu.vue (hunk 5 above)
Comment (Mayank909) [reply src/components/Menu.vue:100]: OK Sir

File: src/components/Menu.vue (hunk 6 above)
Comment (Mayank909) [reply src/components/Menu.vue]: Done Sir

File: src/views/Orders.vue [hunk 7]
```diff
@@ -282,15 +282,23 @@ export default defineComponent({
         });
 
       return jobAlert.present();
+    },
+    fetchJobs(){
+      console.log(' from mounted hook')
```
Comment (ymaheshwari1) [inline src/views/Orders.vue:287]: Unwanted console statement.

File: src/views/Pipeline.vue [hunk 8]
```diff
@@ -388,18 +388,24 @@ export default defineComponent({
...
         });
       } else {
         this.store.dispatch('job/fetchJobHistory', {eComStoreId: this.getCurrentEComStore.productStoreId, viewSize:process.env.VUE_APP_VIEW_SIZE, viewIndex:0, queryString: this.queryString}).then(() => {
           if(event) event.target.complete();
           this.isRetrying = false;
+        }).catch(()=>{
+          this.isRetrying = false;
         });
-      }
+      }
```
Comment (ymaheshwari1) [inline src/views/Pipeline.vue:408]: Remove unwanted change.

**Review by ymaheshwari1 — CHANGES_REQUESTED** [review]

File: src/views/Orders.vue (hunk 7 above)
Comment (Mayank909) [reply src/views/Orders.vue:320]: OK Sir

File: src/views/Pipeline.vue (hunk 8 above)
Comment (Mayank909) [reply src/views/Pipeline.vue:426]: OK Sir

File: src/components/Menu.vue [hunk 9]
```diff
@@ -86,6 +93,16 @@ export default defineComponent({
       userProfile: 'user/getUserProfile'
     })
   },
+  methods: {
+    setEComStore(event: CustomEvent) {
+      if(this.userProfile) {
+        this.store.dispatch('user/setEcomStore', {
+          'eComStore': this.userProfile.stores.find((store: any) => store.productStoreId == event['detail'].value)
```
Comment (ymaheshwari1) [inline src/components/Menu.vue:100]:
```suggestion
          'eComStore': this.userProfile.stores.find((store: any) => store.productStoreId === event['detail'].value)
```

File: src/components/Menu.vue [hunk 10]
```diff
@@ -86,6 +93,16 @@ export default defineComponent({
...
+        })
+      }
+      emitter.emit("selectedShop")
```
Comment (ymaheshwari1) [inline src/components/Menu.vue:103]: I think we can name it something like `changeProductStore` in place of `selectedShop`

**Review by ymaheshwari1 — CHANGES_REQUESTED** [review]

File: src/components/Menu.vue (hunk 9 above)
Comment (Mayank909) [reply src/components/Menu.vue:100]: Ok sir I will add it.

File: src/components/Menu.vue (hunk 10 above)
Comment (Mayank909) [reply src/components/Menu.vue:103]: OK, Sir, I will correct it.

File: src/views/Pipeline.vue [hunk 11]
```diff
@@ -445,16 +445,22 @@ export default defineComponent({
         this.getPendingJobs().then(() => {
           if(event) event.target.complete();
           this.isRetrying = false;
+        }).catch(()=>{
+          this.isRetrying = false;
```
Comment (adityasharma7) [inline src/views/Pipeline.vue:449]: Why we have changed this?

File: src/views/Pipeline.vue [hunk 12]
```diff
@@ -569,11 +575,13 @@ export default defineComponent({
     // setting the current job as empty because when coming back to the pipeline page the currentJob
     // state does not gets updated and hence the job configuration component takes it space in DOM
     await this.store.dispatch('job/updateCurrentJob', { job: {} });
+    emitter.on("productStoreChanged", this.refreshJobs);
```
Comment (adityasharma7) [inline src/views/Pipeline.vue:578]: If we are using refreshJobs, we should handle isRetrying properly. This flag will show spinner on retry button when product store changed, even though user hasn't clicked the button.

**Review by adityasharma7 — CHANGES_REQUESTED** [review]

File: src/views/Pipeline.vue (hunk 12 above)
Comment (Mayank909) [reply src/views/Pipeline.vue:578]: OK, Sir, I will correct it

File: src/views/Pipeline.vue [hunk 13]
```diff
@@ -439,23 +439,23 @@ export default defineComponent({
         event.target.complete();
       })
     },
-    async refreshJobs(event: any) {
-      this.isRetrying = true;
+    async refreshJobs(event: any, retry = false ) {
```
Comment (adityasharma7) [inline src/views/Pipeline.vue:442]:
```suggestion
    async refreshJobs(event: any, isRetrying = false ) {
```

**Review by adityasharma7 — CHANGES_REQUESTED** [review]

File: src/views/Pipeline.vue (hunk 13 above)
Comment (Mayank909) [reply src/views/Pipeline.vue:442]: Ok Sir

**Comment by disha1202** [comment]
Need to update the value of autoCancelDays as well on productStore change
https://github.com/hotwax/job-manager/pull/218

**Pull request merged**